avm/src/daemon/daemon.rs

```diff
@@ -24,6 +24,32 @@ use crate::vm::run::run;
 pub type DaemonResult<T> = std::result::Result<T, Box<dyn Error + Send + Sync>>;
 
 pub static CLUSTER_SECRET: OnceCell<Option<String>> = OnceCell::new();
+pub static DAEMON_PROPS: OnceCell<DaemonProperties> = OnceCell::new();
+
+#[allow(non_snake_case)]
+#[derive(Deserialize, Debug, Serialize)]
+pub struct DaemonFileB64 {
+  pub name: String,
+  pub content: String,
+}
+
+#[allow(non_snake_case)]
+#[derive(Deserialize, Debug, Serialize)]
+pub struct DaemonFilesB64 {
+  pub Dockerfile: Option<DaemonFileB64>,
+  pub appTarGz: Option<DaemonFileB64>,
+  pub envVariables: Option<DaemonFileB64>,
+}
+
+#[allow(non_snake_case)]
+#[derive(Deserialize, Debug, Serialize)]
+pub struct DaemonProperties {
+  pub clusterId: String,
+  pub agzB64: String,
+  pub deployToken: String,
+  pub domain: String,
+  pub filesB64: Option<DaemonFilesB64>,
```

Comment:
so this is weird because the agz is also base64 encoded can we rename `filesB64` to `optionalB64files` or something? This also complicates the data structure because of the options nested within an option.